### PR TITLE
added slice_edge

### LINKED ISSUE #260: Add support for slicing an edge
Slicing an edge with something like

```python
net.slice_edge(edge, start_index, length)
```
Would be very useful for doing parallel contractions using JAX's `pmap`. Each slice of the edge would run on a different device and the final result would just be the sum of all of the slices, which is very easy to do in jax with `psum`.

### PR DESCRIPTION
**Pull request by jensenjhwang**
Resolves #260

### CODE REVIEW

**Comment by jensenjhwang** [comment]
@googlebot I signed it!

File: tensornetwork/tests/network_components_free_test.py [hunk 1]
```diff
@@ -1586,6 +1586,64 @@ def test_split_edge_different_backend_raises_value_error(single_node_edge):
...
+  assert new_edge.axis2 == 1
+  assert new_edge.dimension == 2
+
+  expected_tensor = np.array([[4, 5], [7, 8]])
+  np.testing.assert_allclose(expected_tensor, node.get_tensor())
+
+
+def test_slice_edge_dangling_edge(backend):
+  node = Node(np.arange(9).reshape(3, 3), backend=backend)
+  edge = Edge(node1=node, axis1=0)
```
Comment (Thenerdstation) [inline tensornetwork/tests/network_components_free_test.py:1616]: Just do `edge = node[0]`. Nodes automatically are given dangling edges upon creating.

File: tensornetwork/backends/numpy/numpy_backend.py [hunk 2]
```diff
@@ -37,6 +37,17 @@ def reshape(self, tensor: Tensor, shape: Tensor):
   def transpose(self, tensor, perm):
     return self.np.transpose(tensor, perm)
 
+  def slice(self,
```
Comment (Thenerdstation) [inline tensornetwork/backends/numpy/numpy_backend.py:40]: Be sure to also add this method to the `BaseBackend` abstract class aswell.

**Comment by Thenerdstation** [comment]
Take a look at the linter and pytype errors. Once those are fixed this should be good to pull in.

**Pull request merged**